File: packages/compose/src/main/kotlin/ai/cocso/ui/CCStrings.kt
package ai.cocso.ui

import androidx.compose.runtime.Composable
import androidx.compose.runtime.ReadOnlyComposable
import androidx.compose.ui.res.stringResource

/**
 * The words the views speak, in the user's language.
 *
 * A screen reader read "On", "Off", "Loading" and "Show password" in English
 * to a Korean user, because the views wrote them as literals. These are the
 * module's string resources (English and Korean shipped), so an app adds a
 * language by adding a `values-xx` folder, and a view never carries a word.
 */
object CCStrings {
    @Composable @ReadOnlyComposable fun loading(): String = stringResource(R.string.cc_loading)
    @Composable @ReadOnlyComposable fun progress(): String = stringResource(R.string.cc_progress)
    @Composable @ReadOnlyComposable fun on(): String = stringResource(R.string.cc_on)
    @Composable @ReadOnlyComposable fun off(): String = stringResource(R.string.cc_off)
    @Composable @ReadOnlyComposable fun mixed(): String = stringResource(R.string.cc_mixed)
    @Composable @ReadOnlyComposable fun showPassword(): String = stringResource(R.string.cc_show_password)
    @Composable @ReadOnlyComposable fun hidePassword(): String = stringResource(R.string.cc_hide_password)
    @Composable @ReadOnlyComposable fun close(): String = stringResource(R.string.cc_close)
    @Composable @ReadOnlyComposable fun select(): String = stringResource(R.string.cc_select)
}

File: packages/swiftui/Sources/CocsoUI/CCStrings.swift
import Foundation

/**
 The words the views speak, in the user's language.

 A screen reader read "On", "Off", "Mixed", "Loading" and "Show password" in
 English to a Korean user, because the views wrote them as literals. These are
 keys into the package's `Localizable.strings` (English and Korean shipped), so
 an app adds a language by adding a table, and a view never carries a word.
 */
public enum CCStrings {
    public static var loading: String { localized("cc.loading") }
    public static var progress: String { localized("cc.progress") }
    public static var on: String { localized("cc.on") }
    public static var off: String { localized("cc.off") }
    public static var mixed: String { localized("cc.mixed") }
    public static var showPassword: String { localized("cc.showPassword") }
    public static var hidePassword: String { localized("cc.hidePassword") }
    public static var close: String { localized("cc.close") }
    public static var select: String { localized("cc.select") }
    public static var pagination: String { localized("cc.pagination") }
    public static var previousPage: String { localized("cc.previousPage") }
    public static var nextPage: String { localized("cc.nextPage") }
    public static var breadcrumb: String { localized("cc.breadcrumb") }
    public static var stockSufficient: String { localized("cc.stockSufficient") }
    public static var stockNormal: String { localized("cc.stockNormal") }
    public static var stockInsufficient: String { localized("cc.stockInsufficient") }
    /// "Page 3" — the number placed where the language puts it.
    public static func page(_ number: Int) -> String {
        String(format: localized("cc.page"), locale: .current, Int64(number))
    }

    private static func localized(_ key: String) -> String {
        NSLocalizedString(key, bundle: .module, comment: "")
    }
}
